### DISCUSSION TITLE
java.lang.String to int conversion

### DISCUSSION THREAD

**Discussion by vsquared** [py5coding/py5generator#463] [Q&A] [answered]
How do you convert a _java.lang.String_ to an int (or integer)?  Things I've tried:
Integer.parseInt()
Integer.valueOf()
int()
eval()
str.isdigit()
All of these result in an error which is lengthy but starts out like this:
```
org.jpype.PyExceptionProxy
	at org.jpype.JPypeContext.createException(Unknown Source)
	at org.jpype.proxy.JPypeProxy.hostInvoke(Native Method)
	at org.jpype.proxy.JPypeProxy.invoke(Unknown Source)
	at jdk.proxy2/jdk.proxy2.$Proxy11.actionPerformed(Unknown Source)
```
**Minimal Example:**
```
# Uses Import mode for py5

from java.awt import *
from javax.swing import *

_wndW = 400
_wndH = 200

def myBtnAction(evnt):   
    strID = evnt.getActionCommand()
    print("strID =", strID)
# ==== None of these work as expected =====
#     intValue = Integer.parseInt(strID)
#     intValue = Integer.valueOf(strID)
    intValue = int(strID)
    print(intValue)
        
def setup():
    global frame
    size(_wndW, _wndH)
    wnd = get_surface()
    wnd.set_title('Click on text field, hit Enter.')
    canvas = wnd.get_native()
    frame = canvas.getFrame()
    frame.setBounds(60,100,_wndW, _wndH)
    frame.remove(canvas)
    txtFld = JTextField("")
    txtFld.setBounds(180,60,40,30)
# **** This is what I want to retrieve and convert to an int or integer
    txtFld.setActionCommand(str(42))
    txtFld.addActionListener(myBtnAction)
    frame.add(txtFld)
    txtFld.repaint()
    
```

**Comment by vsquared** [comment, marked as answer]
Never mind; I think I figured out how to do it.  You have to convert  _java.lang.String_  to a python string using _str(cmdStr)_, and then convert that to an int using _int(pyStr)_.

```
# Converts java.lang.String to int
    pyStr = str(cmdStr)
    value = int(pyStr)
    print(type(value))
```

**Reply by hx2A** [reply]
Yes, this is the best option! Glad you figured it out. Converting the java.lang.String with str will convert this to a Python object, and once it is in the Python world you can do what you ordinarily would to convert a str to an int.

The alternatives you initially explored would at best convert the Java String to a Java Integer Object, and that Integer Object will not always play nice with other Python numbers. It gets a bit messy and I don't recommend it.
